File: src/c/magma/LastLocator.c
import magma.option.None;
import magma.option.Option;
import magma.option.Some;
struct LastLocator(String infix) implements Locator {
	Option<Integer> locate(String input){
		auto index = input.lastIndexOf(infix());
		if (index ==  - 1) {
			return None<>();
		}
		return Some<>(index);
	}
	int sliceLength(){
		return this.infix.length();
	}
}

File: src/java/magma/LastLocator.java
package magma;

import java.util.Optional;

public record LastLocator(String slice) implements Locator {
    @Override
    public Optional<Integer> locate(String input) {
        final var index = input.lastIndexOf(this.slice);
        if (index == -1) return Optional.empty();
        return Optional.of(index);
    }

    @Override
    public int length() {
        return this.slice.length();
    }
}